extensions/transfer-functions/transfer-functions-core/src/main/java/org/eclipse/dataspaceconnector/transfer/functions/core/TransferFunctionsCoreServiceExtension.java

```diff
@@ -0,0 +1,177 @@
+
+    private static final String DEFAULT_LOCAL_TRANSFER_URL = "http://localhost:9090/transfer";
+    private static final String DEFAULT_LOCAL_CHECK_URL = "http://localhost:9090/checker";
+
+    private Monitor monitor;
+
+    private boolean localTransfer;
+    private Set<String> protocols;
+    private ServiceExtensionContext context;
+
+    @Override
+    public Set<String> provides() {
+        return Set.of("dataspaceconnector:transfer-function");
+    }
+
+    @Override
+    public void initialize(ServiceExtensionContext context) {
+        this.context = context;
+        monitor = context.getMonitor();
+
+        protocols = getSupportedProtocols(context);
+
+        var transferType = context.getSetting(TRANSFER_TYPE, "HTTP");
+        localTransfer = "LOCAL".equalsIgnoreCase(transferType);
+
+        if (localTransfer) {
+            monitor.info("Local transfer functions are enabled");
+        } else {
+            initializeHttpFunctions(context);
+        }
+
+        monitor.info("Initialized Transfer Functions Core extension");
+    }
+
+    @Override
+    public void start() {
+        if (localTransfer) {
+            var localTransferFunction = context.getService(LocalTransferFunction.class, true);
+            if (localTransferFunction == null) {
+                monitor.severe("Local transfer functions are configured. An implementation of " +
+                        LocalTransferFunction.class.getName() + " must be provided. Transfer functions are disabled.");
+                return;
+            }
+            var flowController = new LocalFunctionDataFlowController(protocols, localTransferFunction);
+            var flowManager = context.getService(DataFlowManager.class);
+            flowManager.register(flowController);
+
+            var statusChecker = new LocalStatusChecker(localTransferFunction);
+            var statusCheckerRegistry = context.getService(StatusCheckerRegistry.class);
+            protocols.forEach(protocol -> statusCheckerRegistry.register(protocol, statusChecker));
+        }
+        monitor.info("Started Transfer Functions Core extension");
+    }
+
+    @Override
+    public void shutdown() {
+        monitor.info("Shutdown Transfer Functions Core extension");
+    }
+
+    private void initializeHttpFunctions(ServiceExtensionContext context) {
+        var httpClient = createHttpClient(context);
+        context.registerService(TransferFunctionInterceptorRegistry.class, httpClient::addInterceptor);
+
+        var typeManager = context.getTypeManager();
+        var transferEndpoint = context.getSetting(TRANSFER_URL_KEY, DEFAULT_LOCAL_TRANSFER_URL);
+        var checkEndpoint = context.getSetting(CHECK_URL_KEY, DEFAULT_LOCAL_CHECK_URL);
+        var configuration = HttpFunctionConfiguration.Builder.newInstance()
+                .transferEndpoint(transferEndpoint)
+                .checkEndpoint(checkEndpoint)
+                .clientSupplier(httpClient::build)
```

Comment:
the `httpClient` can be directly resolved using the context. In that case a `requires` dependecy to `"dataspaceconnector:http-client"` must be defined.

```suggestion
                .clientSupplier(context.getService(OkHttpClient.class)
```

Using one HttpClient globally has the upside of central configuration (timeouts, etc.) and error handling.